desktop_version/src/Game.cpp

```diff
@@ -4926,8 +4973,58 @@ void Game::savestats()
     }
 
     xml::update_tag(dataNode, "controllerSensitivity", controllerSensitivity);
+}
 
-    FILESYSTEM_saveTiXml2Document("saves/unlock.vvv", doc);
+void Game::loadsettings(int* width, int* height, bool* vsync)
+{
+    tinyxml2::XMLDocument doc;
+    if (!FILESYSTEM_loadTiXml2Document("saves/settings.vvv", doc))
+    {
+        savesettings();
+        puts("No settings.vvv found");
+    }
+
+    tinyxml2::XMLHandle hDoc(&doc);
+    tinyxml2::XMLElement* pElem;
+    tinyxml2::XMLHandle hRoot(NULL);
+
+    {
```

Comment:
What's up with this block? Seems like it's missing a return and has some weird indentation... otherwise the rest seems okay.

Reply:
That's just how the other XML files are handled by the game (go look at `Game::loadstats()`, for instance). It's bad because of the weird indentation and the fact that it does error checking only to throw it away, but I'm keeping it for consistency with the rest of the code.

It should probably be fixed, though. And all the XML parsing should probably have error handling (immediate segfault if any element in an XML document is missing!!!).

Reply:
Ah, got it - this is one of those things your brain blocks out after you see it and ship it.